xcm/xcm-builder/src/transactional.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use frame_support::storage::{with_transaction_unchecked, TransactionOutcome};
+use xcm::latest::prelude::*;
+use xcm_executor::traits::ProcessTransaction;
+
+/// Transactional processor implementation using frame transactional layers.
+pub struct FrameTransactionalProcessor;
+impl ProcessTransaction for FrameTransactionalProcessor {
```

Comment:
We don't normally say "Frame" on everything that uses FRAME, right?
```suggestion
pub struct TransactionalProcessor;
impl ProcessTransaction for TransactionalProcessor {
```

Reply:
I think it might be important to specifically mention this in the case of the xcm-executor as it could eventually also be used in implementations that do not use FRAME.